### PR TITLE
Hotfix/entitysearchinpurview

### PR DESCRIPTION
**Pull request by rodrigomonteiro-gbb**
FIx prolbem with the return of the Purview Query API, validating that the returned record is equal to the Qname searched for names with "-" or spaces.

### CODE REVIEW

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 1]
```diff
@@ -267,7 +270,7 @@ public async Task<QueryValeuModel> QueryInPurview(string TypeName)
                     filter.Add("filter", new JObject());
                     ((JObject)filter!["filter"]!).Add("and", new JArray());
                 }
-                //SUpport for the cases when user use wasbs protocol for a ADLS GEN 2
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:270]: Spelling issue

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 2]
```diff
@@ -267,7 +270,7 @@ public async Task<QueryValeuModel> QueryInPurview(string TypeName)
...
+                //SUpport for the cases when user use wasbs protocall for a ADLS GEN 2
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:273]: Spelling issue

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 3]
```diff
@@ -646,25 +638,21 @@ public async Task<EntityModel> search_entities(string QualifiedName, string type
         }
 
         /// <summary>
-        /// Delete unused Placeholder entities with specific Qualified Name
+        /// Delete unused Dumies entities with specific Qualified Name
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:641]: spelling dummies

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 4]
```diff
@@ -745,19 +733,19 @@ private async Task<string> GetToken()
             {
                 // Invalid scope. The scope has to be of the form "https://resourceurl/.default"
                 // Mitigation: change the scope to be as expected
-                _logger.LogError("Error getting Authentication Token for Purview API");
+                _logger.LogError("Error geting Authentication Token ofr Purview API");
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:736]: Spelling 'getting' 'for'

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 5]
```diff
@@ -745,19 +733,19 @@ private async Task<string> GetToken()
...
                 return String.Empty;
             }
             catch (Exception coreex)
             {
 
-                _logger.LogError($"Error getting Authentication Token for Purview API - Invalid scope. The scope has to be of the form {scopes}");
+                _logger.LogError($"Error geting Authentication Token ofr Purview API - Invalid scope. The scope has to be of the form {scopes}");
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:742]: getting / for spelling

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 6]
```diff
@@ -837,20 +815,17 @@ public async Task<JObject> GetByGuid(string guid)
         /// <summary>
         /// Get Entity in Microsoft Purview using bulk/uniqueAttribute/type API
         /// </summary>
-        /// <param name="qualifiedName">Qualified Name to Search by</param>
+        /// <param name="quilifiedName">Qualified Name to Search by</param>
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:818]: spelling qualifiedName

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 7]
```diff
@@ -837,20 +815,17 @@ public async Task<JObject> GetByGuid(string guid)
...
         /// <param name="typeName">Type Name to search by</param>
         /// <returns></returns>
-        public async Task<List<Asset>> Get_Entity(string qualifiedName, string typeName)
+        public async Task<List<Asset>> Get_Entity(string quilifiedName, string typeName)
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:821]: spelling qualifiedName

File: function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs [hunk 8]
```diff
@@ -837,20 +815,17 @@ public async Task<JObject> GetByGuid(string guid)
...
             var correlationId = Guid.NewGuid().ToString();
             var token = await this.GetToken();
 
-            //var baseurl = $"https://{Environment.GetEnvironmentVariable("PurviewAccountName") ?? ""}.{Environment.GetEnvironmentVariable("ResourceUri") ?? "purview.azure.com"}";
-            //var purviewApiEndPoint = Environment.GetEnvironmentVariable("purviewApiEndPoint") ?? "{ResourceUri}/catalog/api";
-            //var purviewApiEntityByTypeMethod = Environment.GetEnvironmentVariable("purviewApiEntityByTypeMethod") ?? "/atlas/v2/entity/bulk/uniqueAttribute/type/";
             var purviewSearchEndpoint = $"{config!.purviewApiEndPoint!.ToString().Replace("{ResourceUri}", config!.PurviewApiBaseUrl())}{config!.purviewApiEntityByTypeMethod!}";
 
-            List<Asset> entities = await PurviewclientHelper.GetEntityFromPurview(correlationId, qualifiedName.Trim('/').ToLower(), purviewSearchEndpoint, token, typeName);
+            List<Asset> entities = await PurviewclientHelper.GetEntityFromPurview(correlationId, quilifiedName.Trim('/').ToLower(), purviewSearchEndpoint, token, typeName);
```
Comment (marktayl1) [inline function-app/adb-to-purview/src/Function.Domain/Helpers/PurviewCustomType.cs:828]: spelling qualifiedName

**Review by wjohnson — APPROVED** [review]
Confirmed that the behavior observed in #50 is resolved with this test.

**Comment by wjohnson** [comment]
@rodrigomonteiro-gbb please review https://github.com/microsoft/Purview-ADB-Lineage-Solution-Accelerator/pull/71/files before merging the changes.  This PR will help us to avoid having to make a change to the configuration documentation.

**Pull request merged**